src/bitmap.cpp

```diff
@@ -106,21 +106,21 @@ Bitmap::Bitmap(const std::string& filename, bool transparent, uint32_t flags) {
 	void* pixels = nullptr;
 
 	uint8_t data[4] = {};
-	size_t bytes = fread(&data, 1, 4, stream);
-	fseek(stream, 0, SEEK_SET);
+	size_t bytes = stream->read(reinterpret_cast<char*>(data),  4).gcount();
+	stream->seekg(0, std::ios::ios_base::beg);
 
 	bool img_okay = false;
 
 	if (bytes >= 4 && strncmp((char*)data, "XYZ1", 4) == 0)
-		img_okay = ImageXYZ::ReadXYZ(stream, transparent, w, h, pixels);
+		img_okay = ImageXYZ::ReadXYZ(*stream, transparent, w, h, pixels);
 	else if (bytes > 2 && strncmp((char*)data, "BM", 2) == 0)
-		img_okay = ImageBMP::ReadBMP(stream, transparent, w, h, pixels);
+		img_okay = ImageBMP::ReadBMP(*stream, transparent, w, h, pixels);
 	else if (bytes >= 4 && strncmp((char*)(data + 1), "PNG", 3) == 0)
-		img_okay = ImagePNG::ReadPNG(stream, (void*)nullptr, transparent, w, h, pixels);
+		img_okay = ImagePNG::ReadPNG(*stream, transparent, w, h, pixels);
 	else
 		Output::Warning("Unsupported image file %s (Magic: %02X)", filename.c_str(), *reinterpret_cast<uint32_t*>(data));
 
-	fclose(stream);
+	stream.reset(); //Deletes the stream and closes the file
```

Comment:
Not strictly needed, unless you have a reason to want to close the file early and release the resources.